src/v/storage/tests/log_segment_appender_test.cc

```diff
@@ -179,14 +185,8 @@ static void run_test_can_append_10MB_sequential_write_sequential_read(
     auto appender = make_segment_appender(f, fallocate_size);
 
     // write sequential. then read all
-    iobuf original;
     constexpr size_t one_meg = 1024 * 1024;
-    {
-        const auto data = random_generators::gen_alphanum_string(1024);
-        for (size_t i = 0; i < 1024 * 10; ++i) {
-            original.append(data.data(), data.size());
-        }
-    }
+    iobuf original = make_random_data(one_meg);
```

Comment:
it kinda looks like the original version populated `original` with 10 MBs in this case?

Reply:
Good catch. This file was all conflicts on rebase. I'll look over it again and update this.

Reply:
Wondering why this test didn't fail. Seems like `make_file_input_stream()` or the following read should fail on the index out of range?

Reply:
hmmm. you're right it does look like it should fail or crash.

Reply:
#4077